client/state/inline-support-article/reducer.js

```diff
@@ -4,13 +4,31 @@ import {
 	SUPPORT_ARTICLE_DIALOG_CLOSE,
 } from 'calypso/state/action-types';
 
+function defaultPostId() {
+	if ( ! window || ! URLSearchParams ) {
+		return null;
+	}
+
+	const searchParams = new URLSearchParams( window.location.search );
+	return searchParams.has( 'support-article' )
+		? parseInt( searchParams.get( 'support-article' ) )
+		: null;
+}
+
+function defaultPostUrl() {
+	const postId = defaultPostId();
+	if ( postId ) {
+		return 'https://support.wordpress.com?p=' + postId;
+	}
+	return null;
+}
+
 export default withStorageKey(
 	'inlineSupportArticle',
 	(
 		state = {
-			postId: null,
-			postUrl: null,
-			isVisible: false,
+			postId: defaultPostId(),
```

Comment:
Reducers are required to be pure functions. By relying on `window`, we're introducing a side effect, and that's forbidden in reducers.